clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -444,10 +444,21 @@ private boolean maybeSendAndPollTransactionalRequest() {
         AbstractRequest.Builder<?> requestBuilder = nextRequestHandler.requestBuilder();
         Node targetNode = null;
         try {
-            targetNode = awaitNodeReady(nextRequestHandler.coordinatorType());
-            if (targetNode == null) {
+            FindCoordinatorRequest.CoordinatorType coordinatorType = nextRequestHandler.coordinatorType();
+            targetNode = coordinatorType != null ?
+                    transactionManager.coordinator(coordinatorType) :
+                    client.leastLoadedNode(time.milliseconds());
+            if (targetNode != null) {
+                awaitNodeReady(targetNode, coordinatorType);
```

Comment:
How to deal with the non-ready targetNode after calling ```awaitNodeReady``` ?

Reply:
@chia7712 Thanks for the review. Good point, added check to retry in that case, also added unit test.